clj-refactor.el

```diff
@@ -1957,6 +1964,61 @@ following this convention: https://stuartsierra.com/2015/05/10/clojure-namespace
     (cljr--call-middleware-sync "namespace-aliases")
     parseedn-read-str))
 
+(defun cljr--call-middleware-suggest-libspec (alias-ref language-context)
+  "Suggest libspec entries for an `alias-ref' in a `language-context'.
+
+Returns a candidate list of libspec entry strings. `alias-ref' is
+a string representing a namespace alias. `language-context' is a
+2 element list representing the language context of the buffer,
+followed by the language context at the current point. Assume
+same context as buffer if context at current point is nil.
+
+Passes through the custom `cljr-magic-require-namespaces' so that
+users can specify default recommended alias prefixes that may not
+appear in the project yet."
+  (seq-let (buffer-context point-context) language-context
```

Comment:
`seq-let` is probably nice but this lib (and I, coincidentally) only use seq.el for "seq stuff", not for other conveniences which would increase cognitive load to casual maintainers.

I'm generally happy with `let*` (and prefer `let*` over `let`, no matter what)

Reply:
I've addressed this to your recommendation. From my perspective, destructuring binding is much easier to follow over manually parsing out car/cadr of each element, but to each their own. I appreciate your concern about cognitive load from unfamiliar constructs, but it's also frustrating and discouraging as a contributor to so frequently encounter code included in emacs core that is then dismissed at review. Do you have a style guide somewhere that I could follow to discover which are recommended and which are discouraged instead of discovering all of these at review time?

Reply:
>  I could follow to discover which are recommended and which are discouraged instead of discovering all of these at review time?

Not that I know of

Generally, afaict in clojure-emacs .el code we lean toward simple constructs. Even then, different reviewers might have different preferences.

My genuine recommendation would be to simply be ready to tweak some things upon request. I myself cannot predict the feedback I'll receive in the variety of repos I contibute to - some little churn on superficial things is just part of life! :)